tests/publiccloud/check_registercloudguest.pm

```diff
@@ -116,27 +109,37 @@ sub run {
     }
 
     $instance->ssh_script_retry(cmd => "sudo ${path}registercloudguest $regcode_param", timeout => 300, retry => 3, delay => 120);
+    check_instance_registered($instance);
+
+    $instance->ssh_script_retry(cmd => "sudo ${path}registercloudguest $regcode_param --force-new", timeout => 300, retry => 3, delay => 120);
+    check_instance_registered($instance);
 
+    register_addons_in_pc($instance);
+
+    set_var('PUBLIC_CLOUD_CHECK_CLOUDREGISTER_EXECUTED', '1');
+}
+
+sub check_instance_registered {
+    my ($instance) = @_;
     if ($instance->ssh_script_output(cmd => 'zypper -x lr | grep "<repo" | wc -l', timeout => 300) == 0) {
         record_info('zypper lr', $instance->ssh_script_output(cmd => 'zypper lr'));
         die('The list of zypper repositories is empty.');
     }
     if ($instance->ssh_script_output(cmd => 'sudo ls /etc/zypp/credentials.d/ | wc -l') == 0) {
         die('Directory /etc/zypp/credentials.d/ is empty.');
     }
+}
 
-    $instance->ssh_script_retry(cmd => "sudo ${path}registercloudguest $regcode_param --force-new", timeout => 300, retry => 3, delay => 120);
-    if ($instance->ssh_script_output(cmd => 'zypper -x lr | grep "<repo" | wc -l', timeout => 300) == 0) {
+sub check_instance_unregistered {
+    my ($instance, $error) = @_;
+    if ($instance->ssh_script_output(cmd => 'zypper lr | grep "<repo" | wc -l', timeout => 300) > 0) {
```

Comment:
```suggestion
    if ($instance->ssh_script_output(cmd => 'zypper -x lr | grep "<repo" | wc -l', timeout => 300) > 0) {
```
May be you missed the `-x` option here ?

Reply:
At that point, if you want, you can also avoid `wc` by using `grep -c "..."`

Reply:
no I haven't missed `-x` here , because this line is NOT replacement for line 129 on the left but for line 103 . Git diff just fooling you here . The one and only thing which I am doing here is I am just wrap **similar function calls** into single function to reduce  code duplication. So I don't want to introduce ANY improvements to this functions on purpose to not move too many pieces simultaneously .

Reply:
Yes ok, but even that L103 was _Wrong_, sorry!
 We cannot filter that output `grep '<repo'`, searching for xml tags, unless you add -x option, triggering xml output.

Reply:
Kind refeerence to output types:
**zypper lr**
```
#  | Alias                                | Name                                            | Enabled | GPG Check | Refresh
---+--------------------------------------+-------------------------------------------------+---------+-----------+--------
 1 | Cloud_Tools_CI                       | Cloud:Tools:CI (openSUSE_Tumbleweed)            | Yes     | (r ) Yes  | No
 2 | Google-Chrome                        | Google-Chrome                                   | Yes     | (r ) Yes  | Yes
...
```

**zypper -x lr** 
```
zypper -x lr 
<?xml version='1.0'?> 
<stream> 
<repo-list> 
<repo alias="Cloud_Tools_CI" name="Cloud:Tools:CI (openSUSE_Tumbleweed)" type="rpm-md" priority="99" enabled="1" autorefresh=
"0" gpgcheck="1" repo_gpgcheck="1" pkg_gpgcheck="0" raw_gpgcheck="1" gpgkey="https://download.opensuse.org/repositories/Cloud
:/Tools:/CI/openSUSE_Tumbleweed/repodata/repomd.xml.key"> 
<url>https://download.opensuse.org/repositories/Cloud:/Tools:/CI/openSUSE_Tumbleweed/</url> 
</repo> 
<repo alias="Google-Chrome" name="Google-Chrome" type="rpm-md" priority="99" enabled="1" autorefresh="1" gpgcheck="1" repo_gp
gcheck="1" pkg_gpgcheck="0">
...
</repo> 
</repo-list> 
</stream>
```
Therefore, only adding `-x` in `zypper -x lr` we can have **xml** output and a filter on xml tags like `...|grep '<repo'` to make sense.

Reply:
ok make sense , will do needed changes

Reply:
fixed in latest version